api/ccapi/include/functions.h

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 SeungBaek Hong <sb92.hong@samsung.com>
+ *
+ * @file   functions.h
+ * @date   15 March 2024
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author SeungBaek Hong <sb92.hong@samsung.com>
+ * @bug	   No known bugs except for NYI items
+ *
+ * @note This is experimental API and not stable.
+ */
+
+#ifndef __ML_TRAIN_TENSOR_H__
+#define __ML_TRAIN_TENSOR_H__
+
+#if __cplusplus < MIN_CPP_VERSION
+#error "CPP versions c++17 or over are only supported"
+#endif // __cpluscplus
+
+#include <iostream>
+#include <memory>
+#include <vector>
+
+namespace ml {
+namespace train {
+
+    class Function;
+
+    // TensorCore for graph configuration
+    class TensorNode {
+    public:
+        bool is_leaf = true;
+        bool requires_grad = false;
+        std::shared_ptr<Function> creator = nullptr;
+
+        TensorNode() {
+            is_leaf = true;
+            requires_grad = false;
+        }
+
+        TensorNode(std::shared_ptr<Function> &func) {
+            is_leaf = false;
+            creator = func;
```

Comment:
Try to follow modern C++ initialization expression:
```C
TensorNore():
    if_leaf(true), requires_grad(false), creator(nullptr) {
    ...
}
```

Reply:
Thanks! I applied it. :)